com.unity.ml-agents/Runtime/Sensors/VectorSensor.cs

```diff
@@ -21,16 +21,20 @@ public class VectorSensor : ISensor, IBuiltInSensor
         /// </summary>
         /// <param name="observationSize">Number of vector observations.</param>
         /// <param name="name">Name of the sensor.</param>
-        public VectorSensor(int observationSize, string name = null)
+        public VectorSensor(int observationSize, string name = null, ObservationType observationType = ObservationType.Default)
         {
-            if (name == null)
+            if (name == null || name == "")
             {
                 name = $"VectorSensor_size{observationSize}";
+                if (observationType != ObservationType.Default)
+                {
+                    name += "_goal";
```

Comment:
```suggestion
                    name += $"_{observationType.ToString()}";
```
(double check that compiles)